### PR TITLE
fix(testing): AuthRule, AuthClassExtension + KafkaBrokerEnvironmentRu…

### PR DESCRIPTION
**Pull request by christopher-cudennec**
…le set environment variables for Java version < 16

This restores the old behavior before PR #1287

### CODE REVIEW

File: sda-commons-server-auth-testing/src/main/java/org/sdase/commons/server/auth/testing/AuthClassExtension.java [hunk 1]
```diff
@@ -5,16 +5,19 @@
...
 import org.junit.jupiter.api.extension.ExtensionContext;
 import org.sdase.commons.server.auth.config.AuthConfig;
+import org.sdase.commons.server.testing.Environment;
 
 public class AuthClassExtension extends AbstractAuth
     implements BeforeAllCallback, AfterAllCallback {
 
   @SuppressWarnings("WeakerAccess")
-  public static final String AUTH_RULE_ENV_KEY = "AUTH_RULE";
+  private static final String AUTH_RULE_ENV_KEY = "AUTH_RULE";
```
Comment (JoergSiebahn) [inline sda-commons-server-auth-testing/src/main/java/org/sdase/commons/server/auth/testing/AuthClassExtension.java:20]: NIT can we remove the warning above?

File: sda-commons-server-auth-testing/src/main/java/org/sdase/commons/server/auth/testing/AuthRule.java [hunk 2]
```diff
@@ -104,30 +107,38 @@ private void init() {
...
       fail("Failed to create the config keys: " + e.getMessage());
     }
   }
 
+  private void initTestAuth(String authConfig) {
+    final TestRule testRule = createTestRule(authConfig);
+    delegate = RuleChain.outerRule(testRule);
+  }
+
+  public static TestRule createTestRule(String authConfig) {
```
Comment (JoergSiebahn) [inline sda-commons-server-auth-testing/src/main/java/org/sdase/commons/server/auth/testing/AuthRule.java:130]: Could we use less visibility to make it internal or is anybody required to use this method apart from the internal tests?

Comment (christopher-cudennec) [reply sda-commons-server-auth-testing/src/main/java/org/sdase/commons/server/auth/testing/AuthRule.java:130]: Package private should also work. Thanks for pointing it out! 👍

File: sda-commons-server-auth-testing/src/main/java/org/sdase/commons/server/auth/testing/AuthClassExtension.java (hunk 1 above)
Comment (christopher-cudennec) [reply sda-commons-server-auth-testing/src/main/java/org/sdase/commons/server/auth/testing/AuthClassExtension.java:20]: Yeah. Just a leftover. Will remove it.

**Review by JoergSiebahn — APPROVED** [review]

**Pull request merged**